src/rollback_to_stable/rts_btree_walk.c

```diff
@@ -114,25 +114,59 @@ __rts_btree_walk_page_skip(
     return (0);
 }
 
+/*
+ * __rts_btree_walk_progress_msg --
+ *     Log a verbose message about the progress of rollback to stable on a btree walk.
+ */
+static void
+__rts_btree_walk_progress_msg(
+  WT_SESSION_IMPL *session, struct timespec rollback_start, uint64_t *rollback_msg_count)
+{
+    struct timespec cur_time;
+    uint64_t time_diff;
+
+    __wt_epoch(session, &cur_time);
+
+    /* Time since the rollback started. */
+    time_diff = WT_TIMEDIFF_SEC(cur_time, rollback_start);
+
+    if ((time_diff / WT_PROGRESS_MSG_PERIOD) > *rollback_msg_count) {
+        __wt_verbose(session, WT_VERB_RECOVERY_PROGRESS,
+          "Rollback to stable has been performing on %s for %" PRIu64
+          " seconds. For more detailed logging, enable WT_VERB_RTS ",
+          session->dhandle->name, time_diff);
+        (*rollback_msg_count) = time_diff / WT_PROGRESS_MSG_PERIOD;
```

Comment:
I fixed a bug where if a single table takes 1 minute, this function would print out 3 times because of the increment. The increment assumes that only 20 seconds has elapsed since last time we called the function.

Reply:
Nice! I would prefer if there was a way to combine this with `__rts_progress_msg` though, since they're almost the same. The difficulty is the rollback count in the other message, perhaps we split the code into the bookkeeping side and the logging side - e.g. `__rts_progress_msg` would become something like

```
uint64_t time_diff;

if (__rts_should_emit_progress_msg(&time_diff, rollback_msg_count))
    __wt_verbose(...);
```

Happy to discuss this offline as well, there's probably something I haven't considered.

Reply:
I have tried to separate the bookkeeping side and it made the code feel more disorganized because we needed to maintain our own time_diff, cur_time and rollback_msg_count now. I have combined the two into one function and then i added a boolean.

Reply:
This is also OK, but if we add the history store back in we'll need a way to disambiguate that case too. Possibly an enum, but don't be afraid to reconsider using  `__rts_should_emit_progress_msg` - we probably don't need to print the time diff (since that'll be clear from the log timestamps) which should help clean it up a bit, and solve the potential lurking enum problem.

Reply:
Nice! We need something similar for compact progress messages (WT-9607). I'll put a note on that ticket suggesting this as a potential way to address the issue there.